src/analyses/base.ml

```diff
@@ -1025,12 +1024,12 @@ struct
     try
       let fp = eval_fv (Analyses.ask_of_ctx ctx) ctx.global ctx.local fval in
       if AD.mem Addr.UnknownPtr fp then begin
-        M.warn "Function pointer %a may contain unknown functions." d_exp fval;
+        M.warn ~category:Imprecise "Function pointer %a may contain unknown functions." d_exp fval;
         dummyFunDec.svar :: AD.to_var_may fp
       end else
         AD.to_var_may fp
     with SetDomain.Unsupported _ ->
-      M.warn "Unknown call to function %a." d_exp fval;
+      M.warn ~category:Unsound "Unknown call to function %a." d_exp fval;
```

Comment:
Yes, in the first case we still at least have some of the functions that may be called, while we even lost that here.

Reply:
Not necessarily, above it might also be a singleton set just containing `UnknownPtr` and `dummyFunDec.svar` in both cases still calls `special`, such that all the same unknown function invalidations happen.

Reply:
I unified behavior now